app/frontend/src/cljs/teet/task/task_view.cljs

```diff
@@ -220,15 +220,123 @@
+            :confirm-button-text  (tr [:buttons :confirm])
+            :cancel-button-text  (tr [:buttons :cancel])
+            :modal-text (tr [:task-part :submit-results-confirm])
+            :close-on-action? true}
+           [buttons/button-primary {:size :small
+                                    :id (str "submit-button-" (:db/id task-part))
+                                    :on-click (e! task-controller/->SubmitTaskPartResults (:db/id task) (:db/id task-part))}
+            (tr [:task-part :submit-for-approval])]]]
+
+          :file.part.status/reviewing
+          [when-authorized :task/review task
+           [:<>
+           [buttons/button-with-confirm
+          {:action (e! task-controller/->ReviewTaskPart(:db/id task) (:db/id task-part) :accept)
+           :modal-title (str (tr [:task-part :approve-part]) "?")
+           :confirm-button-text  (tr [:buttons :confirm])
+           :cancel-button-text  (tr [:buttons :cancel])
+           :modal-text (tr [:task-part :approve-part-confirm])
+           :close-on-action? true}
+          [buttons/button-green {:style {:margin-right "1em"}
+                                 :size :small
+                                 :id (str "accept-button-" (:db/id task-part))
+                                 :on-click (e! task-controller/->ReviewTaskPart (:db/id task) (:db/id task-part) :accept)}
+           (tr [:task-part :accept])]]
+           [buttons/button-with-confirm
+            {:action (e! task-controller/->ReviewTaskPart (:db/id task) (:db/id task-part) :reject)
+            :modal-title (str (tr [:task-part :reject-part]) "?")
+            :confirm-button-text  (tr [:buttons :confirm])
+            :cancel-button-text  (tr [:buttons :cancel])
+            :modal-text (tr [:task-part :reject-part-confirm])
+            :close-on-action? true}
+            [buttons/button-warning {:size :small
+                                     :id (str "reject-button-" (:db/id task-part))
+                                     :on-click (e! task-controller/->ReviewTaskPart (:db/id task) (:db/id task-part) :reject)}
+             (tr [:task-part :reject])]]]]
+
+          :file.part.status/completed
+          [when-authorized :task/reopen-task task
+           [buttons/button-with-confirm
+            {:action (e! task-controller/->ReopenTaskPart (:db/id task) (:db/id task-part))
+             :modal-title (str (tr [:task-part :reopen-part]) "?")
+             :confirm-button-text  (tr [:buttons :confirm])
+             :cancel-button-text  (tr [:buttons :cancel])
+             :modal-text (tr [:task-part :reopen-part-confirm])
+             :close-on-action? true}
+            [buttons/button-primary {:size :small
+                                     :id (str "reopen-button-" (:db/id task-part))
+                                     :on-click (e! task-controller/->ReopenTaskPart (:db/id task) (:db/id task-part))}
+             (tr [:task-part :reopen])]]]
+          [:<>])]])))
```

Comment:
If the only thing that changes is the modal text and the button maybe this could be refactored a bit to not duplicate so much stuff?